### PR TITLE
Added discover tests

### PR DESCRIPTION
**Pull request by islamov089**

### CODE REVIEW

File: src/discover/getDiscover.ts [hunk 1]
```diff
@@ -0,0 +1,55 @@
+import { AxiosResponse } from "axios";
+import { DiscoverDetails } from "./discoverDetails";
+import { httpRequest } from "../httpMethod/httpRequest";
+import dotenv from "dotenv";
+dotenv.config();
+
+const baseUrl = process.env.BASE_URL;
+
+export async function discoverMovie(): Promise<AxiosResponse<DiscoverDetails>> {
```
Comment (jkalandarov) [inline src/discover/getDiscover.ts]: the function name is confusing again.

File: src/discover/getDiscover.ts [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+      queryParams
+    );
+    return response;
+  } catch (error) {
+    console.error("Error fetching account details:", error);
+    throw error;
+  }
+}
+
+export async function discoverTv(): Promise<AxiosResponse<DiscoverDetails>> {
```
Comment (jkalandarov) [inline src/discover/getDiscover.ts]: function name is confusing again.

File: src/discover/getDiscover.ts (hunk 1 above)
Comment (islamov089) [reply src/discover/getDiscover.ts]: done

File: src/discover/getDiscover.ts (hunk 2 above)
Comment (islamov089) [reply src/discover/getDiscover.ts]: fixed

File: src/discover/getDiscover.ts [hunk 3]
```diff
@@ -30,7 +30,7 @@ export async function getMovieInfo(): Promise<AxiosResponse<DiscoverDetails>> {
   }
 }
 
-export async function getTvInfo(): Promise<AxiosResponse<DiscoverDetails>> {
+export async function getTvShowlist(): Promise<AxiosResponse<MovieDetails>> {
```
Comment (jkalandarov) [inline src/discover/getDiscover.ts]: return  type: Promise<AxiosResponse<TvShows>> may be?

Comment (islamov089) [reply src/discover/getDiscover.ts]: nope return type is `Promise<AxiosResponse<MovieDetails>>`

File: src/discover/getDiscover.test.ts [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+  });
+});
+
+describe("Test for Discover Tv api", () => {
+  test("Verify that a account id is not null.", async () => {
+    const data = await getTvShowlist();
+    expect(data.data.id).not.toBe(null);
+    expect(data.data.results[0].id).toBeGreaterThan(50000);
+    expect(data.data.results.length).toBeGreaterThan(0);
+    expect(data.data.results.length).toBe(20);
```
Comment (jkalandarov) [inline src/discover/getDiscover.test.ts]: If results.length will be more or less than 20 your test will fail.

File: src/discover/getDiscover.test.ts [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+
+    expect(status).toBe(200);
+  });
+});
+
+describe("Test for Discover Tv api", () => {
+  test("Verify that a account id is not null.", async () => {
+    const data = await getTvShowlist();
+    expect(data.data.id).not.toBe(null);
+    expect(data.data.results[0].id).toBeGreaterThan(50000);
```
Comment (jkalandarov) [inline src/discover/getDiscover.test.ts]: If sorting is applied on the backend side, and the order of items in the results array changes, your test will fail.

File: src/discover/getDiscover.ts (hunk 3 above)
Comment (jkalandarov) [reply src/discover/getDiscover.ts]: Is your function supposed to get MovieDetails or TvShowsList?

**Review by jkalandarov — CHANGES_REQUESTED** [review]

File: src/discover/getDiscover.test.ts (hunk 5 above)
Comment (islamov089) [reply src/discover/getDiscover.test.ts]: ok

File: src/lists/lists.ts [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+import { AxiosResponse } from "axios";
+import { httpRequest } from "../httpMethod/httpRequest";
+import * as ResponseType from "./listDetails";
+
+dotenv.config();
+
+const sessionId = process.env.SESSION_ID;
+const listId = process.env.LIST_ID;
+
+export async function clearItemsFromList(): Promise<AxiosResponse> {
```
Comment (jkalandarov) [inline src/lists/lists.ts]: from which list does this function clear items?

File: src/lists/lists.ts [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+  const url = `list/${listId}/clear`;
+  const queryParam: any = {
+    session_id: sessionId,
+    confirm: true,
+  };
+
+  try {
+    const response = await httpRequest<AxiosResponse>("POST", url, queryParam);
```
Comment (jkalandarov) [inline src/lists/lists.ts]: httpRequest returns an object with AxioResponse type:
```
response: AxiosResponse {
  config: {},
  headers: {},
  data: {},
  status: 200
}
```
The `AxioResponse` interface you're passing as a type defines the type of `response.data`. In current implementation, you are defining:
```
response.data: AxiosResponse {
  config: {},
  headers: {},
  data: {},
  status: 200
}
```
`httpRequest<AxiosResponse<T>> `
image

**Review by jkalandarov — CHANGES_REQUESTED** [review]

File: src/discover/getDiscover.ts [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+  const queryParams = {
+    include_adult: false,
+    include_video: false,
+    language: "en-US",
+    page: 1,
+    sort_by: "popularity.desc",
+  };
+
+  try {
+    const response = await httpRequest<MovieDetails>(
```
Comment (jkalandarov) [inline src/discover/getDiscover.ts]: Is it supposed to return TvShowList?

File: src/discover/getDiscover.ts [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+  }
+}
+
+export async function getTvShowlist(): Promise<AxiosResponse> {
+  const url = `discover/tv`;
+  const queryParams = {
+    include_adult: false,
+    include_video: false,
+    language: "en-US",
+    page: 1,
```
Comment (jkalandarov) [inline src/discover/getDiscover.ts]: What if I want to get results for page 3?

File: src/httpMethod/httpRequest.ts [hunk 10]
```diff
@@ -24,10 +26,11 @@ export async function httpRequest<T>(
...
         },
+        body: body ? JSON.stringify(body) : undefined,
     };
 
     try {
-        const response = await axios(fullUrl, options);
+        const response = await axios((baseUrl+fullUrl), options);
         return response;
     } catch (error) {
         console.error(`Error making ${method} request to ${fullUrl}:`, error);
```
Comment (jkalandarov) [inline src/httpMethod/httpRequest.ts:36]: No need to use console methods since thrown errors print on console.

File: src/lists/lists.test.ts [hunk 11]
```diff
@@ -0,0 +1,23 @@
+import { clearItemsFromMovieList } from "./lists";
+import { getMovieListDetails } from "./lists";
+
+describe("Test for Lists  API ", () => {
+  test("Check Status code.", async () => {
+    const { status } = await clearItemsFromMovieList();
+    expect(status).toBe(201);
+
```
Comment (jkalandarov) [inline src/lists/lists.test.ts]: 1. First get any item in the movie list and assign it to a variable. <br>
2. Clear the list. <br>
3. Either check the item doesn't exist in the list anymore or check movie list length is equal to zero.

File: src/lists/lists.ts [hunk 12]
```diff
@@ -0,0 +1,45 @@
+import dotenv from "dotenv";
+import { AxiosResponse } from "axios";
+import { httpRequest } from "../httpMethod/httpRequest";
+import * as ResponseType from "./listDetails";
+
+dotenv.config();
+
+const sessionId = process.env.SESSION_ID;
+const listId = process.env.LIST_ID;
```
Comment (jkalandarov) [inline src/lists/lists.ts]: listId is not a sensitive data.

File: src/lists/lists.ts [hunk 13]
```diff
@@ -0,0 +1,45 @@
...
+
+export async function clearItemsFromMovieList(): Promise<AxiosResponse> {
```
Comment (jkalandarov) [inline src/lists/lists.ts]: the function should accept list id as a parameter.

**Review by jkalandarov — CHANGES_REQUESTED** [review]

**Review by jkalandarov — APPROVED** [review]

**Pull request merged**